include/sys/mem_manage.h

```diff
@@ -204,6 +204,75 @@ void *k_mem_map(size_t size, uint32_t flags);
 size_t k_mem_region_align(uintptr_t *aligned_addr, size_t *aligned_size,
 			  uintptr_t addr, size_t size, size_t align);
 
+#ifdef CONFIG_DEMAND_PAGING
+/**
+ * Evict a page-aligned virtual memory region to the backing store
+ *
+ * Useful if it is known that a memory region will not be used for some time.
+ * All the data pages within the specified region will be evicted to the
+ * backing store if they weren't already, with their associated page frames
+ * marked as available for mappings or page-ins.
+ *
+ * None of the associated page frames mapped to the provided region should
+ * be pinned.
+ *
+ * Note that there are no guarantees how long these pages will be evicted,
+ * they could take page faults immediately.
+ *
+ * If CONFIG_DEMAND_PAGING_ALLOW_IRQ is enabled, this function may not be
+ * called by ISRs as the backing store may be in-use.
+ *
+ * @param addr Base page-aligned virtual address
+ * @param size Page-aligned data region size
+ * @retval 0 Success
+ * @retval -ENOMEM Insufficient space in backing store to satisfy request.
+ *         The region may be partially paged out.
+ */
+int z_mem_page_out(void *addr, size_t size);
```

Comment:
if those are app-facing APIs, shouldn't this be mem_page_out(..), mem_page_in(..) etc?

Once we start adding documentation for paging in general and start referencing those APIs it will not work. Any functions starting with z_ will not be processed by doxygen.

Reply:
You should not be concerned about deprecation policy here, those APIs will go in as experimental/unstable and the deprecation policy only applies when this went through 2 releases and when more architectures are supported, so we should be good to move to k_mem_...

Reply:
promoted.